frontend/ui/src/reducers/data/messages/index.ts

```diff
@@ -25,6 +26,63 @@ function mergeMessages(oldMessages: Message[], newMessages: Message[]): Message[
   return sortBy(messages, message => message.sentAt);
 }
 
+const findConversationId = (state: Messages, messageId: string) => {
```

Comment:
Ok this is a funny consequence of the data model I had not anticipated. This means that the update becomes `O(n)` compared to `O(1)` compared to the other stores.

Reply:
But I don't have a better suggestion other than populating a separate field on the store with a lookup table to amortize the cost

Reply:
...or just adding the conversation id in the websocket message. That would make this also a `O(1)` thing again.